
This sucks, but it's better than what OFW was giving us.

diff -ur grub2-1.96+20080601.old/kern/ieee1275/init.c grub2-1.96+20080601/kern/ieee1275/init.c
--- grub2-1.96+20080601.old/kern/ieee1275/init.c	2008-06-08 20:40:07.000000000 +0200
+++ grub2-1.96+20080601/kern/ieee1275/init.c	2008-06-08 22:03:17.000000000 +0200
@@ -84,6 +84,9 @@
     /* We already set prefix in grub_machine_init().  */
     return;

+#ifdef __i386__
+  grub_env_set ("prefix", "(sd,1)/");
+#else
   if (grub_ieee1275_get_property (grub_ieee1275_chosen, "bootpath", &bootpath,
 				  sizeof (bootpath), 0))
     {
@@ -121,6 +124,7 @@

   grub_free (filename);
   grub_free (prefix);
+#endif
 }

 /* Claim some available memory in the first /memory node. */
